### DISCUSSION TITLE
Does all input require interrupts?

### DISCUSSION THREAD

**Discussion by muhammadelmogy** [grblHAL/STM32F4xx#80] [General]
I want to fit some inputs in stm32f411 and It is a little bit tight over there because not all pins can have the interrupt for the same pin number in different port!!

So may be this is obvious but I just want to make sure, does all input require interrupts?
If no, can I have none-required interrupt input on the same pin/port number as the other input with interrupt?
<sub>I have the feeling the answer will be yes!</sub>

**Comment by terjeio** [comment]
>So may be this is obvious but I just want to make sure, does all input require interrupts?

No. Limits and control inputs requires interrupt support, except Z -limit that _can_ be configured for polling. The probe input does not require interrupt support, but it may become an option later (if the pin supports it).

> If no, can I have none-required interrupt input on the same pin/port number as the other input with interrupt?

Yes.

**Comment by muhammadelmogy** [comment]
Thanks, appreciate your reply
